common/models/user.js

```diff
@@ -557,6 +571,9 @@ module.exports = function(User) {
       cb(err);
       return cb.promise;
     }
+    if (options.password && options.password.length > MAX_PASSWORD_LENGTH) {
+      this.validatePassword(options.password);
```

Comment:
On the second thought, this patch is already taking way too long to complete. Let's move the try/catch wrapping out of scope of this work, so that we can get this fix landed soon.